engine/lite/components/request-response-tests/src/test/scala/pl/touk/nussknacker/engine/lite/components/requestresponse/LiteRequestResponseFunctionalTest.scala

```diff
@@ -167,6 +168,48 @@ class LiteRequestResponseFunctionalTest extends AnyFunSuite with Matchers with E
       result shouldBe expected
     }
   }
+
+  test("should handle nested non-raw mode") {
+    val output =
+      """
+        |{
+        |  "type": "object",
+        |  "properties": {
+        |    "field": {
+        |      "type": "object",
+        |      "additionalProperties": {
+        |        "type": "object",
+        |        "properties": {
+        |          "additionalField": { "type": "number" }
+        |        }
+        |      }
+        |    }
+        |  }
+        |}
+        |""".stripMargin
+    val params: List[(String, Expression)] = List(
+      "field" -> """{trala:'lala'}"""
+    )
+    val scenario = ScenarioBuilder
+      .requestResponse("test")
+      .additionalFields(properties = Map(
+        "inputSchema" -> "{}",
+        "outputSchema" -> output
+      ))
+      .source("input", "request")
+      .emptySink(sinkName, "response",
+        (SinkRawEditorParamName -> ("false": Expression)) :: params: _*
+      )
+
+    val result = runner.runWithRequests(scenario) { invoker =>
+      invoker(HttpRequest(HttpMethods.POST, entity = "{}")).rightValue
+    }
+
+    result should matchPattern {
+      case Invalid(NonEmptyList(CustomNodeError("response", _, Some("field")), Nil)) =>
```

Comment:
Can you add checking of some part of message? Because without this I can't see what exactly this test doing. It is 'trala' and should be 'additionalField'?